File: ext/examples/num_gens.rs
//! This prints the number of generators in each $\Ext^{s, n + s}$ in the format `n,s,num_gens`.
use ext::{
    chain_complex::{ChainComplex, FreeChainComplex},
    utils::query_module,
};

fn main() -> anyhow::Result<()> {
    ext::utils::init_logging()?;

    let resolution = query_module(None, false)?;

    for b in resolution.iter_stem() {
        println!(
            "{},{},{}",
            b.n(),
            b.s(),
            resolution.number_of_gens_in_bidegree(b)
        );
    }
    Ok(())
}

File: ext_py/examples/num_gens.py
#!/usr/bin/env python3
"""Print the number of generators in each Ext^{s, n+s} as `n,s,num_gens`.

Python port of ext/examples/num_gens.rs.
"""

import _query as query


def main():
    resolution = query.query_module()

    for b in resolution.iter_stem():
        print(f"{b.n()},{b.s()},{resolution.number_of_gens_in_bidegree(b)}")


if __name__ == "__main__":
    main()
